src/flamenco/runtime/program/fd_bpf_loader_v3_program.h

```diff
@@ -39,11 +26,20 @@ int
 fd_bpf_loader_v3_is_executable( fd_exec_slot_ctx_t * slot_ctx,
                                 fd_pubkey_t const *  pubkey );
 
+/* fd_bpf_loader_v3_program_execute is the shared entry point for bpf 
+   user-defined programs as well as executions of the program itself. */
+
+int
+fd_bpf_loader_v3_program_execute( fd_exec_instr_ctx_t instr_ctx );
+
+/* TODO: add comment here */
+
 fd_account_meta_t const *
 read_bpf_upgradeable_loader_state_for_program( fd_exec_txn_ctx_t * txn_ctx,
```

Comment:
This needs to get refactored and cleaned up, but leaving for now as it's used in other parts of the codebase. It serves the same funciton as `read_bpf_upgradeable_loader_state_for_program`